scripts/mas_sso_config.sh

```diff
@@ -8,155 +9,172 @@ FLEET_OPERATOR_ROLE=kas_fleetshard_operator
-kasClientId=$(jq -r '.[].id' <<< "$KAS")
-
-#/auth/admin/realms/rhoas/clients/de121cf7-a6b2-4d39-a99c-8da787454a66/service-account-user
-SVC=$(curl -sk --header "Content-Type: application/json" --header "Authorization: Bearer $TOKEN" "$KEYCLOAK_URL"/auth/admin/realms/rhoas-kafka-sre/clients/"$kasClientId"/service-account-user)
-svcUserId=$(jq -r '.id' <<< "$SVC")
-echo "$svcUserId"
-
-FINAL=$(curl -sk --data-raw '[{"id": '"$manageUser"',"name": "manage-users"},{"id": '"$manageRealm"',"name": "manage-realm"},{"id": '"$manageClients"',"name": "manage-clients"}]' --header "Content-Type: application/json" --header "Authorization: Bearer $TOKEN" "$KEYCLOAK_URL"/auth/admin/realms/rhoas-kafka-sre/users/"$svcUserId"/role-mappings/clients/"$realmMgmtClientId")
-echo "$FINAL"
-
-REALM=rhoas-kafka-sre
-REALM_URL="${KEYCLOAK_URL}/auth/admin/realms/${REALM}"
-
-FLEETMANAGER_ADMIN_CLIENT=kas-fleet-manager
-FLEETMANAGER_ADMIN_ROLE=kas-fleet-manager-admin-full
-
-# Obtain admin user access token
-RESULT=$(curl -sk --data "grant_type=password&client_id=$CLIENT_ID&username=$KEYCLOAK_USER&password=$KEYCLOAK_PASSWORD" "$KEYCLOAK_URL"$TOKEN_PATH)
-ACCESS_TOKEN=$(jq -r '.access_token' <<< "$RESULT")
-AUTHN_HEADER="Authorization: Bearer ${ACCESS_TOKEN}"
-JSON_CONTENT="Content-Type: application/json"
-
-ADMIN_CLIENT_ID=$(curl --fail --show-error -sk -H"${AUTHN_HEADER}" ${REALM_URL}/clients?clientId=${FLEETMANAGER_ADMIN_CLIENT} | jq -r '.[].id')
-
-# attempt to fetch the role id if role already created
-ADMIN_FULL_ROLE_ID=$(curl --fail --show-error -sk -H"${AUTHN_HEADER}" ${REALM_URL}/roles |\
-    jq -r -c '.[] | select( .name | contains("'${FLEETMANAGER_ADMIN_ROLE}'")).id')
-
-if [ -z $ADMIN_FULL_ROLE_ID ] ; then 
-    # Only create "kas-fleet-manager-admin-full" role if not already created
-    ROLE_CREATE_RESPONSE=$(curl --fail --show-error -sk -H"${JSON_CONTENT}" -H"${AUTHN_HEADER}" ${REALM_URL}/roles --data-raw '{
-    "name": "'${FLEETMANAGER_ADMIN_ROLE}'"
-    }')
-
-    if [ $? -ne 0 ] ; then
-        exit 1
-    fi
-
-    echo "Created role: ${FLEETMANAGER_ADMIN_ROLE}"
-
-    # fetch the created role id
-    ADMIN_FULL_ROLE_ID=$(curl --fail --show-error -sk -H"${AUTHN_HEADER}" ${REALM_URL}/roles |\
-    jq -r -c '.[] | select( .name | contains("'${FLEETMANAGER_ADMIN_ROLE}'")).id')
-fi
-
-echo $ADMIN_FULL_ROLE_ID
 
-if [ $? -ne 0 ] ; then
-    exit 1
-fi
+COUNT=0
+while [[ $COUNT -lt $MAX_KEYCLOAK_ATTEMPT_COUNT ]]; do
+  ((COUNT+=1))
+  if [[ $(curl -s -o /dev/null -w ''%{http_code}'' http://localhost:8180/auth/realms/master) == "200" ]]; then
+    echo "[$COUNT/$MAX_KEYCLOAK_ATTEMPT_COUNT] - Keycloak ready"
+    KEYCLOAK_READY=1
+    break
+  fi
+
+  echo "[$COUNT/$MAX_KEYCLOAK_ATTEMPT_COUNT] - Waiting for keycloak server at localhost:8180"
+  sleep 10
+done
+
+if [[ $KEYCLOAK_READY ]]; then
```

Comment:
Instead of an if-else at that level we can do the opposite:
If it is not ready then we show that error message and we exit with err code (exit 1)
Otherwise we just continue running the rest of the tasks

Reply:
done